narwhal/network/src/connectivity.rs

```diff
@@ -5,8 +5,14 @@ use anemo::PeerId;
 use std::collections::HashMap;
 use tokio::task::JoinHandle;
 
+#[cfg(feature = "metrics")]
+use snarkos_metrics::gauge;
+
 pub struct ConnectionMonitor {
     network: anemo::NetworkRef,
+
+    // Only used with metrics, but not worth the effort to make it conditional.
+    #[allow(dead_code)]
```

Comment:
Happy to go with this for now but maybe add a TODO so we don't forget it in future?